In [1]:
import smtplib

# Note
Don’t enter this example in IDLE; it won’t work because smtp.example.com, FakeEmailToTestPython@hotmail.com, MY_SECRET_PASSWORD are just placeholders. This code is just an overview of the process of sending email with Python.

# Sending Email

In [2]:
test = smtplib.SMTP('smtp-mail.outlook.com', 587) # 587 is smtp settings

In [3]:
print(type(test))

<class 'smtplib.SMTP'>


# Note
If you are not connected to the Internet, Python will raise a socket.gaierror: [Errno 11004] getaddrinfo failed or similar exception

In [4]:
test.ehlo()

(250,
 b'MWHPR14CA0010.outlook.office365.com Hello [208.53.71.170]\nSIZE 157286400\nPIPELINING\nDSN\nENHANCEDSTATUSCODES\nSTARTTLS\n8BITMIME\nBINARYMIME\nCHUNKING\nSMTPUTF8')

# Starting TLS Encryption
If you are connecting to port 587 on the SMTP server (that is, you’re using TLS encryption), you’ll need to call the starttls() method next. This required step enables encryption for your connection. If you are connecting to port 465 (using SSL), then encryption is already set up, and you should skip this step.

In [5]:
test.starttls()

(220, b'2.0.0 SMTP server ready')

starttls() puts your SMTP connection in TLS mode. The 220 in the return value tells you that the server is ready.

# Warning
Be careful about putting passwords in your source code. If anyone ever copies your program, they’ll have access to your email account! It’s a good idea to call input() and have the user type in the password. It may be inconvenient to have to enter a password each time you run your program, but this approach will prevent you from leaving your password in an unencrypted file on your computer where a hacker or laptop thief could easily get it.

In [6]:
# this a fake email fell free to use. Please do not change the password so other can use and learn as well
test.login('crobear21@hotmail.com', 'MY_SECRET_PASSWORD') 

(235, b'2.7.0 Authentication successful')

# Note
Gmail’s Application-Specific Passwords

Gmail has an additional security feature for Google accounts called application-specific passwords. If you receive an Application-specific password required error message when your program tries to log in, you will have to set up one of these passwords for your Python script. Check out the resources at http://nostarch.com/automatestuff/ for detailed directions on how to set up an application-specific password for your Google account.

In [7]:
test.sendmail('crobear21@hotmail.com', 'jacquetrahan88@hotmail.com', '''Subject: So long.
\nDear Alice, so long and thanks for all the fish. Sincerely,Bob''')

{}

I added triple quotes """ message """ to get the body of the email to send correctly. 

Example:
>>> smtpObj.sendmail(' my_email_address@gmail.com ', ' recipient@example.com ',
'Subject: So long.\nDear Alice, so long and thanks for all the fish. Sincerely,
Bob')
{}

The sendmail() method requires three arguments.

* Your email address as a string (for the email’s “from” address)

* The recipient’s email address as a string or a list of strings for multiple recipients (for the “to” address)

* The email body as a string

The start of the email body string must begin with 'Subject: \n' for the subject line of the email. The '\n' newline character separates the subject line from the main body of the email.

The return value from sendmail() is a dictionary. There will be one key-value pair in the dictionary for each recipient for whom email delivery failed. An empty dictionary means all recipients were successfully sent the email.

# Disconnecting from the SMTP Server
Be sure to call the quit() method when you are done sending emails. This will disconnect your program from the SMTP server.

In [8]:
test.quit()

(221, b'2.0.0 Service closing transmission channel')